File: Python/algorithms/factorial.py

def iterative_factorial(n):
    fact = 1
    for i in range(2,n+1):
        fact *=i
    return fact

def recur_factorial(n):
    if n ==1:
        return n
    
    return  recur_factorial(n-1)* n
     
     







if __name__ == "__main__":
    print(iterative_factorial(5))
    print(recur_factorial(5))

File: Typescript/algorithms/factorial.ts
export const iterativeFactorial = (n:number):number=>{
    let fact = 1

    for (let i = 2; i <= n ; i++) {
        fact *= i

    }
    return fact
}


const recurFactorial = (n:number):number=>{
    if (n === 1){
        return n
    }
    
    return recurFactorial(n-1)* n
     
    
    
    
}

console.log(iterativeFactorial(5))
console.log(recurFactorial(5))
